pandas/core/reshape/merge.py

```diff
@@ -2197,11 +2170,23 @@ def injection(obj: ArrayLike):
             else:
                 # We get here with non-ndarrays in test_merge_by_col_tz_aware
                 #  and test_merge_groupby_multiple_column_with_categorical_column
-                lbv = flip(left_by_values)
-                rbv = flip(right_by_values)
-                lbv = ensure_object(lbv)
-                rbv = ensure_object(rbv)
-
+                mapped = [
+                    _factorize_keys(
+                        left_by_values[n],
+                        right_by_values[n],
+                        sort=False,
+                        how="outer",
```

Comment:
Not super familiar with this code but does hard coding `"outer"` not have any impact? Wondering if there was a case where two dataframes get joined with very few matches

Reply:
Good point, we can use how=left here. Updated.